Allow URI-references for Security Requirements (3.2.0) (#3821)
OAI/OpenAPI-Specification: versions/3.2.0.md

#### <a name="securityRequirementObject"></a>Security Requirement Object

Lists the required security schemes to execute this operation.
The name used for each property MUST either correspond to a security scheme declared in the [Security Schemes](#componentsSecuritySchemes) under the [Components Object](#componentsObject), or be the URI of a Security Scheme Object.

> Property names that match the syntax of a component name under the Components Object MUST be treated as a component name.

Reviewer: ```suggestion
Property names identical to a component name under the Components Object MUST be treated as a component name.
```

Author: @ralfhandl this word choice was intentional to ensure that any given string is unambiguously resolved as either a component or a URI, and cannot change its resolution based on whether a component exists (that's a problem with `mapping` today, which we actually considered fixing in a PR before deciding that we would not make breaking changes, even small ones, in 3.2 and closed the PR).

The next line shows how to use a URI that would, in the most normalized form, be interpeted as a component name.

This is also something of a a compatibility thing to me (although I'll grant this point is debatable)- keys that always and only resolved as component names in 3.1 should continue to always and only resolve that way.  Only previously illegal keys should have the new behavior.

Reviewer: Ah, so any property name matching `^[a-zA-Z0-9\.\-_]+$` MUST be treated as a security scheme name, and processing MUST fail (or use a default/fallback security scheme) if there is nothing inside `components.securitySchemes` with this name?

Author: @ralfhandl correct.  If there is default/fallback behavior then we definitely need to keep the syntax partitioned so that still works (it shouldn't switch to trying to treat the component name as a path segment, because if it did and by some chance there happened to be a file of that name in the same directory, then when you updated to 3.2 it would start referencing that file instead of falling back).
